pkg/api/types.go

```diff
@@ -476,6 +478,18 @@ type GlusterfsVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// MetadataVolumeSource represents an host directory mounted on a POD. It stores pod's metadata info
+type MetadataVolumeSource struct {
+	// PodNameFileName is the name of the file where name of the pod is written
+	PodNameFileName string `json:"podNameFileName,omitempty"`
```

Comment:
This ad-hoc collection of things stands in stark contrast to the model being evolved as part of #6739 which uses generic(ish) JSON paths to extract fields.

@pmorie @smarterclayton @bgrant0607 @lavalamp 

Can we unify these?  They are both facets of the "downward API".  Could we arrive at something like:

```
type MetadataVolumeSource struct {
    Data map[string]string
}
```

so that what you have here could be replaced with

```
{"podname": "$.metadata.name", "podns": "$.metadata.namespace", "labels": "$.metadata.labels"}
```